modin/pandas/iterator.py

```diff
@@ -15,21 +15,21 @@
 
 
 class PartitionIterator(Iterator):
-    def __init__(self, query_compiler, axis, func):
+    def __init__(self, df, axis, func):
         """PartitionIterator class to define a generator on partitioned data
 
         Args:
-            query_compiler: Data manager for the dataframe
+            df: The dataframe to iterate over
             axis: axis to iterate over
             func: The function to get inner iterables from
                 each partition
         """
-        self.query_compiler = query_compiler
+        self.df = df
         self.axis = axis
         self.index_iter = (
-            iter(self.query_compiler.columns)
+            zip([slice(None)] * len(self.df.columns), iter(self.df.columns))
```

Comment:
why do we need `zip` from `None` slices and column iterator if further in code we access only last field of that tuple?
```python
if self.axis == 1:
    df = self.df[key[-1]]
```

Reply:
Yes it is a good question. I have changed the code to use `iloc`.